src/dash-adapter.js

```diff
@@ -553,7 +539,11 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
    */
   detachMediaSource(): void {
     if (this._shaka) {
-      this._lastTimeDetach = this.currentTime;
+      if (parseInt(this.currentTime) === parseInt(this.duration)) {
```

Comment:
why do we need to parseInt? They are both numbers, no?

Reply:
Right but they're floats could use also Math.round instead

Reply:
Yes, i think it is better. ParseInt expects string.
Did it run flow?

Reply:
I see that it passed flow. Anyway it is more clearer to math round. Someone might think like me it is was done by mistake

Reply:
Sure, it was also in use before - take a look on the line 543 that removed

Reply:
Sounds reasonable then

Reply:
Actually with round will take 5.5 to 6 and parseInt to 5 but I'll use floor instead to make it clear

Reply:
Yes, I meant Math.Floor of course

Reply:
Super edge case that I see now is 800.1 and 799.9 which return false so I added handling for this as well

Reply:
The worst case scenario is that the user will see the end of the playback and click replay, no?

Reply:
So Actually Math.round is better, no?

Reply:
Same for this solution

Reply:
I wonder if we need the floor or keep it simple